### PR TITLE
Replace strings with interpolated strings where relevant across codebase

### PR DESCRIPTION
**Pull request by halgab**
Contributes to #8203 

After this PR, I think we've covered the whole codebase, The next step is adding custom interpolated string handlers at specifc places.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8727)

### CODE REVIEW

**Review by dmitrii-drobotov — APPROVED** [review]
Thank you!

File: src/System.Windows.Forms/src/System/Windows/Forms/CreateParams.cs [hunk 1]
```diff
@@ -72,26 +70,7 @@ public class CreateParams
...
-            builder.Append(", ");
-            builder.Append(Y);
-            builder.Append(", ");
-            builder.Append(Width);
-            builder.Append(", ");
-            builder.Append(Height);
-            builder.Append('}');
-            builder.Append('}');
-            return builder.ToString();
+            return $"CreateParams {{'{ClassName}', '{Caption}', 0x{Style:x}, 0x{ExStyle:x}, {{{X}, {Y}, {Width}, {Height}}}}}";
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/CreateParams.cs:73]: If I am not mistaken I think there are some extra curly braces here
```suggestion
            return $"CreateParams {'{ClassName}', '{Caption}', 0x{Style:x}, 0x{ExStyle:x}, {{X}, {Y}, {Width}, {Height}}}";
```

File: src/System.Windows.Forms/src/System/Windows/Forms/BindingSource.cs [hunk 2]
```diff
@@ -477,13 +477,15 @@ private static string BuildSortString(ListSortDescriptionCollection sortsColln)
                 return string.Empty;
             }
 
-            StringBuilder sb = new StringBuilder(sortsColln.Count);
+            StringBuilder sb = new StringBuilder();
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/BindingSource.cs:480]:
```suggestion
            StringBuilder sb = new();
```

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/FlowLayout.cs [hunk 3]
```diff
@@ -22,23 +22,17 @@ internal partial class FlowLayout : LayoutEngine
...
+            Debug.WriteLineIf(CompModSwitches.FlowLayout.TraceInfo,
+                $"FlowLayout::Layout(container={container}, displayRect={container.DisplayRectangle}, args={args})");
             Debug.Indent();
 #endif
 
             // ScrollableControl will first try to get the layoutbounds from the derived control when
             // trying to figure out if ScrollBars should be added.
             CommonProperties.SetLayoutBounds(container, TryCalculatePreferredSize(container, container.DisplayRectangle, /* measureOnly = */ false));
-#if DEBUG
+
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Layout/FlowLayout.cs:33]: Curious, why did we removed this?

File: src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.WindowClass.cs [hunk 4]
```diff
@@ -101,28 +100,19 @@ internal static WindowClass Create(string? className, WNDCLASS_STYLES classStyle
...
-                    s_currentAppDomainHash + versionSuffix));
-                b.Append(s_currentAppDomainHash);
-                b.Append(versionSuffix);
+                    $"{s_currentAppDomainHash}{versionSuffix}"));
+
+                // While we don't have multiple AppDomains any more, we'll still include the information
+                // to keep the names in the same historical format for now.
 
-                return b.ToString();
+                return $"{Application.WindowsFormsVersion}.{className}.app.0.{s_currentAppDomainHash}{versionSuffix}";
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/NativeWindow.WindowClass.cs:115]: Nice! :)

File: src/System.Windows.Forms/src/System/Windows/Forms/CreateParams.cs (hunk 1 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/CreateParams.cs:73]: In the original string there are 2 curly braces. Here it's interpolated so I have to double them to escape them, right? So 4 curly braces, plus the one that wraps the variable. Here is a small [repro](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYAMBYAUBgRjzwBEB7CAIwBsBTAAgAlaBLAcwAth6BeeggHQBWOAG48AZWBgWUNvQDC1AIYBnVQDllAWwZ8ARADEyZfePwEAnAAoAJPoVhay4LQAKysDtX0A3r4ByXyU1TR1aAF8AmD9mdi4IxMT9AEpRIA)

I agree it doesn't read well. I considered changing it to a raw string, but I liked that it was a one liner.

File: src/System.Windows.Forms/src/System/Windows/Forms/Layout/FlowLayout.cs (hunk 3 above)
Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/Layout/FlowLayout.cs:33]: It was only wrapping a call to `Debug.Unindent` which is already evicted in release

File: src/System.Windows.Forms/src/System/Windows/Forms/CreateParams.cs (hunk 1 above)
Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/CreateParams.cs:73]: Ah yes it was my mistake 😄 , apologies

**Review by lonitra — APPROVED** [review]
🚀

**Pull request merged**